secret_santa.exs

```diff
@@ -0,0 +1,35 @@
+# The code is largely inspired by this tweet: https://twitter.com/gausby/status/1200462062733987841
+
+defmodule EnumUtils do
+  def rotate_left([h | t]) do
+    t ++ [h]
+  end
+end
+
+defmodule SecretSanta do
+  def pair([]) do
+    {:error, "No participants, no secret santas!"}
+  end
+
+  def pair([_]) do
+    {:error, "Alone for Christmas day? Watch 'Home Alone' 1 and 2!"}
+  end
+
+  def pair([_ | _] = participants) do
```

Comment:
I prefer using`is_list()`, though I don't think it changes anything:
```suggestion
  def pair(participants) when is_list(participants) do
```

Reply:
I prefer using `[_|_]` because for me it is clear that the list cannot be empty. With the `is_list()` function you should look at the rest of the code to know if another function manage the empty list case.

The spec are not the same:
```elixir
@spec pair(nonempty_maybe_improper_list) :: {:ok, [{any, any}]}
@spec pair(maybe_improper_list) :: {:ok, [{any, any}]}
```

Maybe it is just a matter of choice :wink:

Reply:
> I prefer using [_|_] because for me it is clear that the list cannot be empty. 

Your list already cannot be empty since you're covering that case in another function :-P.

> Maybe it is just a matter of choice

it is, indeed.